packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerOptions.kt

```diff
@@ -1,51 +1,9 @@
 package expo.modules.documentpicker
 
-import expo.modules.core.Promise
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
 
-data class DocumentPickerOptions(val copyToCacheDirectory: Boolean, val types: Array<String>) {
-  companion object {
-    fun optionsFromMap(options: Map<String, Any?>, promise: Promise): DocumentPickerOptions? {
-      if (options.containsKey("type") && options["type"] == null) {
-        promise.reject("ERR_INVALID_OPTION", "type must be a list of strings")
-        return null
-      }
-      var mimeTypes = arrayListOf("*/*")
-      options["type"]?.let {
-        val types = it as ArrayList<*>
-        if (types.isEmpty() || types[0] !is String) {
-          promise.reject("ERR_INVALID_OPTION", "type must be a list of strings")
-          return null
-        } else {
-          @Suppress("UNCHECKED_CAST")
-          mimeTypes = types as ArrayList<String>
-        }
-      }
-      val copyToCacheDirectory = options["copyToCacheDirectory"]?.let {
-        if (it is Boolean) {
-          return@let it
-        }
-        promise.reject("ERR_INVALID_OPTION", "copyToCacheDirectory must be a boolean")
-        return null
-      } ?: true
-      return DocumentPickerOptions(copyToCacheDirectory, mimeTypes.toTypedArray())
-    }
-  }
-
-  override fun equals(other: Any?): Boolean {
-    if (this === other) return true
-    if (javaClass != other?.javaClass) return false
-
-    other as DocumentPickerOptions
-
-    if (copyToCacheDirectory != other.copyToCacheDirectory) return false
-    if (!types.contentEquals(other.types)) return false
-
-    return true
-  }
-
-  override fun hashCode(): Int {
-    var result = copyToCacheDirectory.hashCode()
-    result = 31 * result + types.contentHashCode()
-    return result
-  }
-}
+data class DocumentPickerOptions(
+  @Field val copyToCacheDirectory: Boolean,
+  @Field val type: List<String>
```

Comment:
```suggestion
  @Field @IsNotEmpty val type: List<String>
```
To ensure that list isn't empty, you can use the `IsNotEmpty` annotation from https://github.com/expo/expo/blob/767aea9caa8cc49fdbf2be727b350fdceaca4358/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/records/Validators.kt#L46
The code will automatically check if the collection has at least one item.

Reply:
Sorry, I had seen these 😅